src/math_util.rb

```diff
@@ -0,0 +1,43 @@
+class MathUtil
```

Comment:
What was the reasoning to put this in a MathUtil since this has operations on arrays of numbers?

Reply:
I wasn't inspired, I think it should be named ArrayUtil.